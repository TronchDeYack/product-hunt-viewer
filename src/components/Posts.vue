<template>
  <div class="posts" v-if="posts">
    <PostCard v-for="(edge, index) in posts.edges" :key="index"
      :img="edge.node.thumbnail.url"
      :votes="edge.node.votesCount"
      :description="edge.node.description"
      :title="edge.node.name" :id="edge.node.id"
    />
  </div>
</template>

<style lang="scss" scoped>
.posts {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 80%;
  margin: 0 auto;
  justify-content: center;
}
</style>

<script>
import PostCard from '@/components/PostCard.vue';

export default {
  name: 'posts',
  components: {
    PostCard,
  },
  props: {
    posts: Object,
  },
};
</script>
