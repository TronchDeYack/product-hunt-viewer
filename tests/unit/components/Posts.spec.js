import { shallowMount } from '@vue/test-utils';

import Posts from '@/components/Posts.vue';

describe('Posts', () => {
  it('Should render.', () => {
    const wrapper = shallowMount(Posts);

    expect(wrapper).toMatchSnapshot();
  });
});
