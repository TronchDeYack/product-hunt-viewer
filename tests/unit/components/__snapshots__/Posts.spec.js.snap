// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Should render. 1`] = `
<div class="posts">
  <postcard-stub img="https://ph-files.imgix.net/85b28898-f9b2-40ec-8856-20be8bf3d2e0?auto=format&amp;fit=crop" title="Kite compositor" votes="731" description="This is a post about interesting things... I guess !"></postcard-stub>
  <postcard-stub img="https://ph-files.imgix.net/85b28898-f9b2-40ec-8856-20be8bf3d2e0?auto=format&amp;fit=crop" title="Kite compositor" votes="731" description="This is a post about interesting things... I guess !"></postcard-stub>
  <postcard-stub img="https://ph-files.imgix.net/85b28898-f9b2-40ec-8856-20be8bf3d2e0?auto=format&amp;fit=crop" title="Kite compositor" votes="731" description="This is a post about interesting things... I guess !"></postcard-stub>
  <postcard-stub img="https://ph-files.imgix.net/85b28898-f9b2-40ec-8856-20be8bf3d2e0?auto=format&amp;fit=crop" title="Kite compositor" votes="731" description="This is a post about interesting things... I guess !"></postcard-stub>
  <postcard-stub img="https://ph-files.imgix.net/85b28898-f9b2-40ec-8856-20be8bf3d2e0?auto=format&amp;fit=crop" title="Kite compositor" votes="731" description="This is a post about interesting things... I guess !"></postcard-stub>
  <postcard-stub img="https://ph-files.imgix.net/85b28898-f9b2-40ec-8856-20be8bf3d2e0?auto=format&amp;fit=crop" title="Kite compositor" votes="731" description="This is a post about interesting things... I guess !"></postcard-stub>
</div>
`;
